src/components/Image/index.tsx

```diff
@@ -37,29 +37,59 @@ function Image({source: propsSource, isAuthTokenRequired = false, session, onLoa
         },
         [onLoad, updateAspectRatio],
     );
+
+    /**
+     * trying to figure out if the current session is expired or fresh from a necessary reauthentication
+     */
+    const previousSessionAge = useRef<number | undefined>();
+    const validSessionAge: number | undefined = useMemo(() => {
+        if (session?.creationDate) {
+            if (previousSessionAge.current) {
+                if (Math.abs(previousSessionAge.current - session.creationDate) < 60000) {
+                    return session.creationDate;
+                }
+                return previousSessionAge.current;
+            }
+            if (Math.abs(new Date().getTime() - session.creationDate) >= CONST.SESSIONS_MAXIDLE_NB_HOURS * 3600000) {
```

Comment:
If we used milliseconds for `CONST.SESSIONS_MAXIDLE_NB_HOURS` we can do this instead:
```suggestion
            if (Math.abs(new Date().getTime() - session.creationDate) >= CONST.SESSIONS_MAXIDLE_NB_HOURS) {
```